subprojects/process-services/src/main/java/org/gradle/process/internal/health/memory/CGroupMemoryInfo.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.process.internal.health.memory;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.io.Files;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+
+public class CGroupMemoryInfo implements OsMemoryInfo {
```

Comment:
Would it also be possible to backport this to Gradle 7? We've started migrating to Gradle 8, but we have over 100 plugins written over 10 years and 1000s of repos, so it takes us a good long while. Without the backport I'm going to have continue to build my plugin that swaps out the `OsMemoryInfo` service using reflection as a stopgap for Gradle 7, which would be a real pain.

Reply:
It's definitely possible.  I'll see what we can do.

Reply:
Sorry to resurrect an old thread but was this every backported? I can't find it in the release notes for 7.6.2 but just want to confirm.

Reply:
Unfortunately, this got lost in the noise and didn't get included in 7.6.2.  I've made sure it's on the list to be included in 7.6.3.